### PR TITLE
Rule contribution page: nuget packages

### PR DESCRIPTION
**Pull request by ashingtondc**
*Issue #, if available:*
#58 
*Description of changes:*
Completed the rule contribution page for NuGet packages, Validation is not implemented yet. 
*Testing done:*
Manual
## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [ x ] I have read the [CONTRIBUTING](https://github.com/aws/porting-assistant-dotnet-ui/blob/develop/CONTRIBUTING.md) doc
- [ x ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/porting-assistant-dotnet-ui/blob/develop/CONTRIBUTING.md#commit-your-change)
- [ x ] I have updated any necessary documentation, including READMEs and comments (where appropriate)

#### Tests

- [ x ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ x ] I have checked that my tests are not configured for a specific environment

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -57,3 +57,4 @@ cli_build
 cli_output
 *.js.map
 ts-dist
+packages/electron/build-scripts/porting-assistant-config.dev.json
```
Comment (flywanli) [inline .gitignore:60]: why you ignore this file

Comment (ashingtondc) [reply .gitignore:60]: I did not want to overwrite the version in the develop branch, which is missing the endpoint info. Should I remove it from gitignore?

File: packages/react/src/components/Feedback/PackageRuleContribution.tsx [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+  packageVersionLatest: boolean;
+  targetFramework: TargetFramework;
+  comments?: string;
+}
+
+const PackageRuleContributionInternal: React.FC<Props> = ({ email, source }) => {
+  const cachedTargetFramework = window.electron.getState("targetFramework");
+  const currentTargetFramework =
+    cachedTargetFramework.id === "netstandard2.1"
+      ? {
```
Comment (flywanli) [inline packages/react/src/components/Feedback/PackageRuleContribution.tsx:45]: no need to filter netstartnd2.1 here

File: packages/react/src/components/Feedback/PackageRuleContribution.tsx [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+      packageVersion: packageVersion,
+      packageVersionLatest: useLatestPackageVersion,
+      targetFramework: {
+        id: targetFramework.value,
+        label: targetFramework.label
+      },
+      comments: comments
+    };
+    // TODO: Actually submit this for verification
+    console.log(submission);
```
Comment (flywanli) [inline packages/react/src/components/Feedback/PackageRuleContribution.tsx:82]: remove log

Comment (flywanli) [reply packages/react/src/components/Feedback/PackageRuleContribution.tsx:82]: use goBack()

File: packages/react/src/components/Feedback/PackageRuleContribution.tsx [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    history.go(-1);
```
Comment (flywanli) [inline packages/react/src/components/Feedback/PackageRuleContribution.tsx:83]: use goBack()

File: packages/react/src/components/Feedback/PackageRuleContribution.tsx [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+  const [targetFramework, setTargetFramework] = useState(currentTargetFramework);
+  const [comments, setComments] = useState("");
+
+  const onSelectTargetFramework = useCallback(([e]) => {
+    setTargetFramework(e.detail.selectedOption);
+    return e.detail.selectedOption;
+  }, []);
+
+  const onCancel = () => {
+    history.go(-1);
```
Comment (flywanli) [inline packages/react/src/components/Feedback/PackageRuleContribution.tsx:67]: use goBack()

File: .gitignore (hunk 1 above)
Comment (flywanli) [reply .gitignore:60]: plz remove, once you merge it. It would overwrite the main branch

Comment (ashingtondc) [reply .gitignore:60]: removed

File: packages/react/src/components/Feedback/PackageRuleContribution.tsx (hunk 2 above)
Comment (ashingtondc) [reply packages/react/src/components/Feedback/PackageRuleContribution.tsx:45]: removed

File: packages/react/src/components/Feedback/PackageRuleContribution.tsx (hunk 3 above)
Comment (ashingtondc) [reply packages/react/src/components/Feedback/PackageRuleContribution.tsx:82]: removed

File: packages/electron/build-scripts/porting-assistant-config.dev.json [hunk 6]
```diff
@@ -7,9 +7,9 @@
     }
   },
   "PortingAssistantMetrics": {
-    "InvokeUrl": "https://localhost",
+    "InvokeUrl": "https://8q2itpfg51.execute-api.us-east-1.amazonaws.com/beta",
```
Comment (flywanli) [inline packages/electron/build-scripts/porting-assistant-config.dev.json:10]: do not commit this secret message

File: packages/react/src/components/Feedback/PackageRuleContribution.tsx [hunk 7]
```diff
@@ -0,0 +1,191 @@
...
+
+interface PackageContribution {
+  packageName: string;
+  packageVersion: string;
+  packageVersionLatest: boolean;
+  targetFramework: TargetFramework;
+  comments?: string;
+}
+
+const PackageRuleContributionInternal: React.FC<Props> = ({ source, email, pagePath }) => {
```
Comment (flywanli) [inline packages/react/src/components/Feedback/PackageRuleContribution.tsx:42]: try to get rid of email and pagePath, it could read from existing value.

**Review by flywanli — APPROVED** [review]

**Pull request merged**